### ISSUE TITLE
MD013: linting & formatting issues when a markdown task has a long URL

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#436]
### rumdl version

0.1.23

### Installation method

Pre-built binary

### Bug description

Long URLs in a markdown task `- [ ] ` results in a diagnostic, and running the formatter makes the originally valid markdown task invalid.

### Steps to reproduce

```md
## Original
- [ ] [some article](https://stackoverflow.blog/2020/11/25/how-to-write-an-effective-developer-resume-advice-from-a-hiring-manager/)

## after formatting
- []
  [some article](https://stackoverflow.blog/2020/11/25/how-to-write-an-effective-developer-resume-advice-from-a-hiring-manager/)

## In comparison, URL in list is not affected
- [some article](https://stackoverflow.blog/2020/11/25/how-to-write-an-effective-developer-resume-advice-from-a-hiring-manager/)
```

### Expected behavior

- no diagnostic warning
- formatting should not make the task invalid

### Actual behavior

- diagnostic warning
- formatting makes the markdown task invalid

### Configuration

```toml
[global]
line-length = 80
```

### Operating system

macOS

### Additional context

Potentially related to various line-length related issues like #393 occurring since 0.1.16

**Comment by rvben** [comment]
Thanks @chrisgrieser for this issue. This should now be resolved (in 0.1.24). Reflow is a tricky beast to get right and keep right for all usecases.
